### ISSUE TITLE
Fix S2583 FP: Rule assumes AddRange is not empty

### ISSUE THREAD

**Issue by manfred-brands** [SonarSource/sonar-dotnet#9503]
### Description

Rule S2583 assumed that `AddRange` always adds at least one element when in reality an enumeration can be empty.

Code does an `AddRange` to a `List` or `ImmutableArray.Builder` then checks for `Count > 0`.

### Repro steps

```csharp
public List<int> AddToList(IEnumerable<int> elements)
{
    var list = new List<int>();
    list.AddRange(elements);

    if (list.Count > 0)
    {
        return list;
    }

    throw new InvalidOperationException("Require more than one element");
}
```

Even calling `list.AddRange(Array.Empty<int>());` is enough to trigger the issue.

### Expected behavior

No issue raised.

### Actual behavior

![image](https://github.com/SonarSource/sonar-dotnet/assets/8655218/089b06d3-27cf-4739-9c6e-7d84f981fc3a)

### Known workarounds

Suppress message

### Related information

NuGet package `SonarAnalyzer.CSharp` Version `9.27.0.93347`
.NET 8.0 SDK
Visual Studio 17.10.3
Windows 10
